### ISSUE TITLE
Glob implementation doesn't match LSP spec

### ISSUE THREAD

**Issue by michaelpj** [bubba/lsp-test#77]
`System.FilePath.Glob` doesn't support `{x,y}` for alternatives, which the LSP spec says are [allowed in glob patterns](https://microsoft.github.io/language-server-protocol/specification#workspace_didChangeWatchedFiles).

```
*Language.Haskell.LSP.Test System.FilePath.Glob> match (compile "{a,b}") "a"
False
```

**Comment by bubba** [comment]
Indeed, I was being lazy but eventually we should implement a globber which implements the spec exactly. Should probably implement it in https://github.com/alanz/lsp
